pmlc/runtime/runtime_registry.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020, Intel Corporation
+
+#include "pmlc/runtime/runtime_registry.h"
+
+#include <list>
+#include <mutex>
+#include <stdexcept>
+#include <utility>
+
+#include "pmlc/util/logging.h"
+#include "llvm/Support/FormatVariadic.h"
+
+namespace pmlc::runtime {
+namespace {
+
+struct LoaderList {
```

Comment:
Can you explain the design of the `Loader`? My thinking was that we could just statically initialize runtimes and have the problem of which runtimes are available be a linker issue.